### DISCUSSION TITLE
How to setup standard linetypes for a loaded DXF file?

### DISCUSSION THREAD

**Discussion by LavenderGu** [mozman/ezdxf#265] [Q&A] [answered]
Hello,
I have a problem with linetypes, when I try to add some lines in an existing dxf file.  How can I use the created standard linetypes in ezdxf when calling ezdxf.readfile()?

Thank you!
Best Regards!

**Comment by mozman** [comment, marked as answer]
```python
import ezdxf
from ezdxf.tools.standards import setup_linetypes

doc = ezdxf.readfile(...)

# create standard linetypes:
setup_linetypes(doc)

# create your own linetype:
doc.linetypes.new('YOUR_LINETYPE', dxfattribs={pattern: [...]})

```
See standard pattern definitions are here:
https://github.com/mozman/ezdxf/blob/c2a9c162038201f9e68296d75378534d731a655a/src/ezdxf/tools/standards.py#L264

[Tutorial](https://ezdxf.mozman.at/docs/tutorials/linetypes.html) how to create linetypes.

**Comment by LavenderGu** [comment]
Thanks for your help!
